isthmus/src/test/java/io/substrait/isthmus/SubstraitRelNodeConverterTest.java

```diff
@@ -150,6 +151,63 @@ public void emit() {
       var relNode = converter.convert(root.getInput());
       assertRowMatch(relNode.getRowType(), R.I32, N.STRING);
     }
+
+    @Test
+    public void leftJoin() {
+      final List<Type> joinTableType = List.of(R.STRING, R.FP64, R.BINARY);
+      final Rel joinTable = b.namedScan(List.of("join"), List.of("a", "b", "c"), joinTableType);
+
+      Plan.Root root =
+          b.root(
+              b.project(r -> b.fieldReferences(r, 0, 1, 3),
+                  b.remap(0, 1, 3),
```

Comment:
One small change. Could you switch these remaps to `6, 7, 8`.

Semantically, if you're referencing fields `0, 1, 3` the project is redundant in these plans, whereas `6, 7, 8` would correspond to the references added in the project so the project would not be redundant. `6,7,8` still triggers the underlying assertion error.

Reply:
I made that change and also ran `./gradlew spotlessApply`.